cdap-master/src/main/java/co/cask/cdap/data/tools/MDSUpgrader.java

```diff
@@ -0,0 +1,347 @@
+    this.appMDS = Transactional.of(executorFactory, new Supplier<DatasetContext<Table>>() {
+      @Override
+      public DatasetContext<Table> get() {
+        try {
+          Table table = DatasetsUtil.getOrCreateDataset(dsFramework, Id.DatasetInstance.from
+                                                          (Constants.DEFAULT_NAMESPACE_ID, Joiner.on(".").join(
+                                                            Constants.SYSTEM_NAMESPACE, DefaultStore.APP_META_TABLE)),
+                                                        "table", DatasetProperties.EMPTY,
+                                                        DatasetDefinition.NO_ARGUMENTS, null);
+          return DatasetContext.of(table);
+        } catch (Exception e) {
+          LOG.error("Failed to access {} table", Joiner.on(".").join(Constants.SYSTEM_NAMESPACE,
+                                                                     DefaultStore.APP_META_TABLE), e);
+          throw Throwables.propagate(e);
+        }
+      }
+    });
+    appStreams = Sets.newHashSet();
+  }
+
+  @Override
+  public void upgrade() throws Exception {
+    appMDS.executeUnchecked(new TransactionExecutor.
+      Function<DatasetContext<Table>, Void>() {
+      @Override
+      public Void apply(DatasetContext<Table> configTable) throws Exception {
+        byte[] appMetaRecordPrefix = new MDSKey.Builder().add(AppMetadataStore.TYPE_APP_META).build().getKey();
+        Scanner rows = configTable.get().scan(appMetaRecordPrefix, Bytes.stopKeyForPrefix(appMetaRecordPrefix));
+        Row row;
+        while ((row = rows.next()) != null) {
+          ApplicationMeta appMeta = GSON.fromJson(Bytes.toString(row.get(COLUMN)), ApplicationMeta.class);
+          appSpecHandler(appMeta.getId(), appMeta.getSpec());
+          appMetaUpgrader(row);
+        }
+        return null;
+      }
+    });
+
+    // Scans for all the streams in the meta table
+    appMDS.executeUnchecked(new TransactionExecutor.
+      Function<DatasetContext<Table>, Void>() {
+      @Override
+      public Void apply(DatasetContext<Table> configTable) throws Exception {
+        byte[] appMetaRecordPrefix = new MDSKey.Builder().add(AppMetadataStore.TYPE_STREAM).build().getKey();
+        Scanner rows = configTable.get().scan(appMetaRecordPrefix, Bytes.stopKeyForPrefix(appMetaRecordPrefix));
+        Row row;
+        while ((row = rows.next()) != null) {
+          StreamSpecification streamSpec = GSON.fromJson(Bytes.toString(row.get(COLUMN)), StreamSpecification.class);
+          // if this stream was not a part of any application
+          if (!appStreams.contains(streamSpec.getName())) {
+            streamHandler(streamSpec);
+          }
+        }
+        return null;
+      }
+    });
+  }
+
+  /**
+   * Upgrades the stream record in the meta table
+   *
+   * @param streamSpec the {@link StreamSpecification} of the stream
+   */
+  private void streamHandler(StreamSpecification streamSpec) {
```

Comment:
methods should be verbs, not nouns.  Also, is there a reason we need a separate method for this if it's just one line?